### DISCUSSION TITLE
Using OpenStreetMap Data

### DISCUSSION THREAD

**Discussion by ianvg** [openmobilityfoundation/curb-data-specification#101] [General]
First time post here, but are there any resources/discussions about how to use this specification with OpenStreetMap data?

**Comment by schnuerle** [comment]
What sort of integration were you thinking of? Somehow defining some of the Curbs API in OSM? Or pulling OSM data into Curbs API?

There hasn't been discussion about this, but the closest thing in the spec is the external [location reference object](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/curbs#location-reference). This could be used to reference some OSM elements related to defined CDS curb places.

**Reply by ianvg** [reply]
My question is fairly broad, so maybe both. The closest specification that has been implemented in OSM I can think of is the GTFS for public transportation by Google. The webpage on OSM discussing how this specification is used OSM and vice versa is [here](https://wiki.openstreetmap.org/wiki/GTFS). I know that there are currently more than one (competing?) specifications for curbs, but this community seems more active and detailed than some so that's why I posted here. Functionally I am interested in what would be the most efficient way to map parking for this specification in OSM and then balance that with OSM end-users current needs and pre-existing typical mapping practices within the OSM community.

**Reply by ianvg** [reply]
I could try and help answer questions related to OSM parking if that's of any help. I'd like for this specification to be able to utilize opendata (e.g. OSM data) if possible and would pike to help in any way I can. Mainly mg ability to help would be to help summarize current parking mapping techniques in OSM and also possibly introduce a new parking schema, if it is found that one would be needed. I'd like to see that OSM parking data is being utilized as efficiently as possible.

**Comment by ianvg** [comment]
Is there anybody that would be interested in this kind of integration? Or is this specification geared for using data from municipalities' private datasets?

**Comment by schnuerle** [comment]
I'm sorry I've been meaning to reply again! 

I think certainly there could be some sort of integration. It would likely be one way, from the public CDS Curbs API endpoints to be put into OSM in some way. I don't think it would go the other way, from OSM into CDS that a city publishes, since CDS can change frequently (depending on how cities implement it). 

For GTFS, only a subset of it is used by OSM, just stops and/or routes. Possibly a subset of CDS Curbs /curbs/* could be used, like info about curb location, geographic polygon shape, parking place shape, and a few of the rules that are supported by OSM. These might need a `cds_` prefix, like the `gtfs_` prefix. Note that the geometries in CDS don't define the actual line of a physical curb, but more the area where policies apply using vehicle GPS, so the geometry shape could be on the street, overlapping the curb onto the sidewalk, or both, so [this `kerb` tag](https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dkerb) does not apply.

So far CDS does not define other items at curbs (like benches, hydrants, planters, trash cans, bike locks, curb cuts/ramps, street light, parking meters, signs, etc), but there is talk of this in the future, so these could be things to bring into OSM later and reconcile with what is there already.

I don't think CDS Events or CDS Metrics would be relevant with OSM.

Noting this [blog post here](https://sharedstreets.io/openstreetmap-and-curb-regulations/) from the defunct SharedStreets talking about Kerb/Curb and OSM with some good thoughts and links.

I'd like to see some integration or data ingestion happen, but would need some OSM experts to lead the way here. Also, so far there is no example public /curbs/zones, /curbs/areas, /curbs/spaces, /curbs/policies feeds yet.
